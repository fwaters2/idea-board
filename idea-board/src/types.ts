export interface Idea {
  id: string;
  title: string;
  description: string;
  created: Date;
  updated?: Date;
}

export enum SortDirection {
  ASC = "ASC",
  DESC = "DESC",
}

export enum SortBy {
  CREATED = "CREATED",
  TITLE = "TITLE",
}

export enum Status {
  SUCCESS = "SUCCESS",
  ERROR = "ERROR",
  IDLE = "IDLE",
  LOADING = "LOADING",
}
