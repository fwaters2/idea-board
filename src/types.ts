export interface Idea {
  id: string;
  title: string;
  description: string;
  created: Date;
  updated?: Date;
}

export interface AppState {
  ideas: Idea[];
  sortKey: SortKey;
  sortDirection: SortDirection;
}

export type SortDirection = "asc" | "desc";

export type SortKey = "created" | "title";

export type Status = "success" | "error" | "idle" | "loading";
