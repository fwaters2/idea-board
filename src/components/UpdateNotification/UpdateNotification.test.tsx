import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { UpdateNotification } from ".";
import { Status } from "../../types";

describe("UpdateNotification component", () => {
  let wrapper: ShallowWrapper<
    React.ReactNode,
    Readonly<{}>,
    React.Component<{}, {}, React.ReactNode>
  >;
  const status: Status = "success";
  const props = { status };

  beforeEach(() => {
    wrapper = shallow(<UpdateNotification {...props} />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render the GreenCircleWithCheckSvg component if status is SUCCESS", () => {
    expect(wrapper.find("GreenCircleWithCheckSvg").exists()).toBe(true);
  });

  it("should render the Loader component if status is not SUCCESS", () => {
    wrapper.setProps({ status: "loading" });
    expect(wrapper.find("Loader").exists()).toBe(true);
  });
});
