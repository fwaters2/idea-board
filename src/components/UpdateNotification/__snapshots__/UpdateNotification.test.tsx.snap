// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateNotification component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
}
`;
